### DISCUSSION TITLE
meanOrientation recalculation in "merge()"

### DISCUSSION THREAD

**Discussion by MaelAllard** [mtex-toolbox/mtex#730] [Ask Anything] [answered]
Hi MTex community,

I was wandering what is exactly doing the "merge()" function by:
`[grainsMerged,parentId] = merge(grains,gB,'calcMeanOrientation')`

I tried it for plagioclase, it works nicely, and I would like to know if the meanOrientation is calculated by considering each twin as a unity, or if their size is also integrated? Or simply if the orientation of the larger twin domain is set as the meanOrientation (it seems not however).

Thank you,
Maël

**Comment by kilir** [comment, marked as answer]
Hi Mael,
if you invoke it just like that, it takes all the mean orientations inside your merged grains and computes a new mean orientation. For twinned grains, this is mostly quite nonsense - the reason why it's not turned on by default. However, you can supply a function handle which takes the grains to be merged as input and compute whatever reasonable measure you like.
Hope that helps.
Cheers,
Rüdiger

**Reply by MaelAllard** [reply]
Hi Rüdiger,
It does help, thank you !

Cheers,
Maël

**Reply by kilir** [reply]
Here's an example  to assign the orientation of the largest parent grain:
```
mtexdata twins
%% select tree grains separated by a tb
[grains, ebsd.grainId]=ebsd.calcGrains;
ebsd = ebsd(grains('id',[66. 75 86]))
[grains, ebsd.grainId]=ebsd.calcGrains('boundary', 'tight',0.8);

%% find the twin boundary
CS = ebsd('M').CS
gB_MgMg = grains.boundary('M','M')

twinning = orientation.map(Miller(0,1,-1,-2,CS),Miller(0,-1,1,-2,CS),...
  Miller(2,-1,-1,0,CS),Miller(2,-1,-1,0,CS))

isTwinning = angle(gB_MgMg.misorientation,twinning) < 5*degree;
twinBoundary = gB_MgMg(isTwinning)

% plot(grains)
% hold on
% plot(twinBoundary,'linecolor','r')
% hold off
%% compare merging 
mg = merge(grains,twinBoundary,'calcMeanOrientation')

h = [Miller(0,0,0,1,CS),Miller(1,1,-2,0,CS)]

% plot of the initial orientations
plotPDF(grains.meanOrientation,grains.grainSize, h)

nextAxis
% plot of "silly" meanOrientation
plotPDF(mg.meanOrientation,h)

mg = merge(grains,twinBoundary,'calcMeanOrientation',@oriLargest)

nextAxis
% plot of meanOrientation being the one of the largest child 
plotPDF(mg.meanOrientation,h)

function mo = oriLargest(g)
    [~,idlargest] = max(g.grainSize);
    mo = g(idlargest).meanOrientation;
end

```

**Reply by MaelAllard** [reply]
Hi Rüdiger,

Thank you for this alternative computation and for the function!

Cheers,
Maël
